### ISSUE TITLE
Consultation on the discrepancies in reproducing results.

### ISSUE THREAD

**Issue by HongliLiu1** [SuhoPark0706/FCP#3]
After training on the COCO dataset (folds 0, 1, 2), I noticed that the test results differ significantly from those presented in the paper. The paper reports 46.4, 56.4, and 55.3, but using the parameters PROMPT_LOSS=1.0, mask_loss=0.5, attn_loss=0.05, attn_drop_out=0.2, and num_layers=7, I obtained results of 33.19, 52.47, and 44.68, respectively.

For the Pascal dataset, I observed results of 66.44 (-8.46), 74.11 (-3.29), 67.76 (-4.04), and 63.95 (-4.85), which also seem to differ quite a bit. I was wondering if you might have used different parameters for COCO and Pascal datasets, or if there could be any steps or settings I might have missed?

Additionally, I noticed that some of the results you shared correspond to the validation set, while others align with the test set. I would be grateful for any clarification on this.

I’ve conducted two rounds of experiments to verify these outcomes, and I truly appreciate the invaluable insights your work has provided. It’s an outstanding piece of research, and I would be very grateful for any further guidance you could offer.

⬇️Below are the results I obtained for COCO fold 0.
*** Training [@Epoch 49] Avg L: 2.52038  mIoU: 55.72   FB-IoU: 73.14   ***
*** Training [@Epoch 49] mIoU: 32.59 | precision: 58.58 | recall: 39.37***

[Epoch: 49] [Batch: 0001/0125] L: 2.49265  Avg L: 2.49265  mIoU: 23.32  |  FB-IoU: 74.85
-> mIoU:  0.23 | precision: 10.73 | recall:  0.24
[Epoch: 49] [Batch: 0051/0125] L: 2.82048  Avg L: 2.81267  mIoU: 35.08  |  FB-IoU: 63.21
-> mIoU:  0.69 | precision: 20.65 | recall:  0.73
[Epoch: 49] [Batch: 0101/0125] L: 2.99989  Avg L: 2.83362  mIoU: 33.05  |  FB-IoU: 63.00
-> mIoU:  0.69 | precision: 20.03 | recall:  0.74

*** Validation [@Epoch 49] Avg L: 2.84837  mIoU: 32.05   FB-IoU: 62.51   ***
*** Validation [@Epoch 49] mIoU:  0.68 | precision: 19.74 | recall:  0.72***

**Comment by SuhoPark0706** [comment]
The number of layers (num_layers) should be 2 or 3.
Thank you

**Comment by HongliLiu1** [comment]
> The number of layers (num_layers) should be 2 or 3. Thank you

I have also conducted experiments with num_layers=3, and the results on the COCO dataset were worse. Specifically, for fold: 0, the performance was not as expected. Could you kindly suggest if there are any other parameter settings that I should consider adjusting?
*** Training [@Epoch 49] Avg L: 1.78612  mIoU: 55.75   FB-IoU: 73.01   ***
*** Training [@Epoch 49] mIoU: 34.42 | precision: 59.42 | recall: 42.53***

[Epoch: 49] [Batch: 0001/0063] L: 1.99844  Avg L: 1.99844  mIoU: 22.04  |  FB-IoU: 67.07
-> mIoU: 10.23 | precision: 18.13 | recall: 16.81
[Epoch: 49] [Batch: 0051/0063] L: 2.05629  Avg L: 2.02273  mIoU: 36.53  |  FB-IoU: 65.16
-> mIoU: 21.08 | precision: 42.61 | recall: 26.21

*** Validation [@Epoch 49] Avg L: 2.03763  mIoU: 35.90   FB-IoU: 64.56   ***
*** Validation [@Epoch 49] mIoU: 20.82 | precision: 41.97 | recall: 26.14***
Thank you for your guidance.

**Comment by SuhoPark0706** [comment]
You should see the best mIoU not the mIoU at the last epoch.
![Image](https://github.com/user-attachments/assets/e689d395-4387-41ed-992d-ac37525f99ec)

And, I recommends you using the scripts I had provoided (scripts/train_pascal.sh, scripts/train_coco.sh)
whose attn_drop_out is set to 0.5

But it seems too low miou...
Here is the docker for your reproduction.
docker pull lakepark/vrpsam:latest

**Comment by jarch-ma** [comment]
How long does it take to train coco? I tried the default setting and it takes nearly a day for one epoch for a single GPU.

**Comment by SuhoPark0706** [comment]
That is because the model extracts SAM masks.
Extracting all masks takes approximately one day on an RTX 3090.

Therefore, you should train your model for one additional day becuase it take one day for training coco.
